Userland/Libraries/LibCore/Stream.cpp

```diff
@@ -165,6 +165,31 @@ ErrorOr<size_t> File::read(Bytes buffer)
     return nread;
 }
 
+ErrorOr<ByteBuffer> File::read_all()
+{
+    off_t file_size = 0;
+    struct stat st;
+    int rc = fstat(m_fd, &st);
+    if (rc == 0)
+        file_size = st.st_size;
+
+    ByteBuffer data;
+    data.ensure_capacity(file_size);
+
+    size_t total_read {};
+    while (true) {
+        auto chunk = TRY(data.get_bytes_for_writing(4096));
```

Comment:
My example contained a magic number, but it's better to move this to a parameter on `read_all` (with a default).